### ISSUE TITLE
Railgun in Weapon_Package_QuakeType crashes when fired

### ISSUE THREAD

**Issue by Elletra** [ClayHanson/BlocklandRebuilt-IssueBoard#298]
### Bug Description
When the railgun in [Weapon_Package_QuakeType](https://bl.kenko.dev/Add-ons/Retail/Weapon/Weapon_Package_QuakeType.zip) is fired, it freezes the game and then crashes.

### To Reproduce
Load a server with the add-on and fire the railgun. **Make sure you install the required `Emote_Critical` add-on too.**

### Metadata:
 - v1.0

### Additional context
So I did a console trace and it's crashing here:
```
Entering quakeRailgunImage::onAfterbeam2(850, 13003, 0)
   Entering drawRaylinequakeRailgun2(92.5529 -125.608 287.594, 91.8823 -132.625 286.312)
      Entering eulerToAxis(0 100.308 -95.4588)
      Leaving eulerToAxis() - return -0.629604 -0.572305 0.52542 2.25045
      Entering onServerAssetLoadWhileInGame(13377)
         Entering createUiNameTable()
         Leaving createUiNameTable() - return 1008
      Leaving onServerAssetLoadWhileInGame() - return 1008
   Leaving drawRaylinequakeRailgun2() - return 13377
```
It seems to be trying to hotload an asset when the railgun is fired. Why is that happening? Well, let's look at `drawRaylinequakeRailgun2()`:

```cs
...
//create it
%shape = new TSStatic() {
	shapeName = "add-ons/weapon_package_quaketype/getter_beam.dts";
};
...
```
So it's creating a `TSStatic` with the `getter_beam.dts` model.

But the problem is, this model should already be loaded on the server, because it's loaded in the `ProjectileData`:

```cs
datablock ProjectileData(RaycastModelLaserProjectile : gunProjectile)
{
   projectileShapeName = "./getter_beam.dts";
   explosion           = quakeRailgunExplosion;
};
```
Even if I add it with `addFileToManifest("add-ons/weapon_package_quaketype/getter_beam.dts")`, it still tries to hotload it.

**Comment by Elletra** [comment]
this add-on has some of the worst code i've ever seen btw

**Comment by ClayHanson** [comment]
I am not getting a crash when firing this weapon, but it could be due to a fix I recently made. Marking as addressed.

**Comment by Elletra** [comment]
It's still crashing for me

**Comment by ClayHanson** [comment]
I tried a couple more times and eventually replicated this in a dedicated server

**Comment by Elletra** [comment]
It's still crashing.

Are you using a fresh, vanilla install or are you using the debug build? Make sure you download the build directly and just install the add-on and `Emote_Critical` and then fire the railgun.

The problem seems to be that it's trying to transmit the `TSStatic` model, even though it's already been added to the manifest... or at least should have been.

**Comment by ClayHanson** [comment]
Even on a vanilla release install, I cannot reproduce this crash in v1.2. Are you sure you've updated both the server & client to the latest version of BLR?
Additionally, you can check if something is in the manifest by using the 'isKnownFromManifest' function.
`echo(isKnownFromManifest("add-ons/weapon_package_quaketype/getter_beam.dts"));`

**Comment by Elletra** [comment]
Yes, I'm using a fresh install every time. `isKnownFromManifest` returns true... This is so bizarre. If anyone else is reading this, are you getting this issue too?

**Comment by Elletra** [comment]
So I think I found a related issue? Every time I add a new file to the manifest, it regenerates the entire manifest and freezes the game for like 30 seconds.
